dbt_subprojects/hourly_spellbook/models/_project/safe/ethereum/safe_ethereum_balances.sql

```diff
@@ -0,0 +1,95 @@
+{{
+    config(
+        schema = 'safe_ethereum',
+        alias = 'balances',
+        partition_by = ['day'],
+        materialized = 'incremental',
+        incremental_strategy = 'merge',
+        file_format = 'delta',
+        unique_key = ['day', 'blockchain', 'address', 'token_address'],
+        post_hook = '{{ expose_spells(\'["ethereum"]\',
+                                    "project",
+                                    "safe",
+                                    \'["safeintern"]\') }}'
+    )
+}}
+
+with changed_balances as (
+    select
+        a.blockchain,
+        day,
+        a.address,
+        token_symbol,
+        token_address,
+        token_standard,
+        token_id,
+        balance,
+        lead(cast(day as timestamp)) over (partition by token_address, a.address, token_id order by day asc) as next_update_day
+    from {{ source('tokens_ethereum', 'balances_daily_agg') }} a
+    join (
+        select
+            address
+            , blockchain
+        from {{ ref('safe_ethereum_safes') }} s
+        where blockchain = 'ethereum'
+    ) q on q.address = a.address
+    where day >= date('2021-07-01')
+        and token_standard in ('native', 'erc20')
+        {% if is_incremental() %}
+        and {{ incremental_predicate('day') }}
+        {% endif %}
+),
+days as (
+    select *
+    from unnest(
+        sequence(cast('2021-07-01' as date), date(date_trunc('day', now())), interval '1' day)
+    ) as foo(day)
+),
+forward_fill as (
+    select
+        blockchain,
+        cast(d.day as timestamp) as day,
```

Comment:
can we actually cast this as `date`?
we've had issues with using `timestamp` type on columns that are used as partitions